### PR TITLE
chore: improve logger config and error codes

### PR DESCRIPTION
**Pull request by behnazh-w**

### CODE REVIEW

**Comment by tromai** [comment]
There is a warning on the Python documentation on the exit codes - https://docs.python.org/3/library/os.html#os._exit
The documentation says:

> Note
> 
> Some of these may not be available on all Unix platforms, since there is some variation. These constants are defined where they are defined by the underlying platform. 

Does it mean that we need to first check whether those constants are available before we use it?

**Comment by behnazh-w** [comment]
> Does it mean that we need to first check whether those constants are available before we use it?

Not really. We are expecting Macaron to work on Linux and MacOS. I'll adjust the classifier to reflect that.

**Comment by behnazh-w** [comment]
> I'll adjust the classifier to reflect that.

See commit [27cbd73](https://github.com/oracle-samples/macaron/pull/174/commits/27cbd7352d23bad335bbe9a80588ff81eb75f444).

**Review by tromai — APPROVED** [review]

File: src/macaron/__main__.py [hunk 1]
```diff
@@ -265,27 +266,39 @@ def main() -> None:
...
 
-    # Set logging debug level. We only need to set for the root logger.
+    if os.path.isdir(args.output_dir):
+        logger.info("Setting the output directory to %s", args.output_dir)
+    else:
+        logger.info("No directory at %s. Creating one ...", args.output_dir)
+        os.makedirs(args.output_dir)
+
+    # Add file handler to the root logger. Remove stream handler from the
+    # root logger to prevent dependencies printing logs to stdout.
```
[the remark this answers is not in the document]
Comment (behnazh-w) [reply src/macaron/__main__.py:290]: > By `dependencies`, do you mean the Python third-party dependencies that we have for Macaron?

Yes, for example SQLAlchemy.

**Pull request merged**